packages/auth/e2e/multiFactor.e2e.js

```diff
@@ -0,0 +1,448 @@
+const {
+  clearAllUsers,
+  getLastSmsCode,
+  createUserWithMultiFactor,
+  createVerifiedUser,
+  getRandomPhoneNumber,
+  signInUserWithMultiFactor,
+} = require('./helpers');
+
+const TEST_EMAIL = 'test@example.com';
+const TEST_PASS = 'test1234';
+
+describe('multi-factor', function () {
+  beforeEach(async function () {
+    await clearAllUsers();
+    await firebase.auth().createUserWithEmailAndPassword(TEST_EMAIL, TEST_PASS);
+    if (firebase.auth().currentUser) {
+      await firebase.auth().signOut();
+      await Utils.sleep(50);
+    }
+  });
+  it('has no multi-factor information if not enrolled', async function () {
+    await firebase.auth().signInWithEmailAndPassword(TEST_EMAIL, TEST_PASS);
+    const { multiFactor } = firebase.auth().currentUser;
+    multiFactor.should.be.an.Object();
+    multiFactor.enrolledFactors.should.be.an.Array();
+    multiFactor.enrolledFactors.length.should.equal(0);
+    return Promise.resolve();
+  });
+  it('can not reuse session after logout', async function () {
+    const { email, password } = await createUserWithMultiFactor();
+
+    try {
+      await firebase.auth().signInWithEmailAndPassword(email, password);
+    } catch (e) {
+      e.message.should.equal(
+        '[auth/multi-factor-auth-required] Please complete a second factor challenge to finish signing into this account.',
+      );
+      const resolver = firebase.auth.getMultiFactorResolver(firebase.auth(), e);
+      const oldHints = resolver.hints[0];
+      const oldSession = resolver.session;
+
+      await firebase.auth().signInWithEmailAndPassword(TEST_EMAIL, TEST_PASS);
+      firebase.auth().currentUser.email.should.equal(TEST_EMAIL);
+
+      try {
+        await firebase.auth().verifyPhoneNumberWithMultiFactorInfo(oldHints, oldSession);
+      } catch (e) {
+        e.message.should.equal(
+          '[auth/invalid-multi-factor-session] No resolver for session found. Is the session id correct?',
+        );
+        return Promise.resolve();
+      }
+
+      return Promise.resolve();
+    }
+
+    return Promise.reject();
+  });
+
+  describe('sign-in', function () {
+    it(':android: requires multi-factor auth when enrolled', async function () {
```

Comment:
Let me see what the output looks like after the e2e runs, as long as there is still evidence in the output that there is a test but it is skipped (and the total test count / skipped test count registers them), that's the real goal for me. How they are skipped isn't so important

Reply:
It does not appear to show up in list of skipped tests (it would be an entry with a `-`) so I don't think this works well - it hides them too well. I'd prefer the if-platform/else-skip style

```
  multi-factor
    ✔ has no multi-factor information if not enrolled
    ✔ can not reuse session after logout (1035ms)
    enroll
      ✔ throws an error for unknown sessions
      ✔ throws an error for unknown verification code
```
https://github.com/invertase/react-native-firebase/actions/runs/3271818247/jobs/5385708987#step:23:282

Reply:
Done :)